File: src/test/java/aoc2015/day07/Day07Test.java
package aoc2015.day07;

import static aoc.InputFetcherKt.loadExample;
import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

class Day07Test {

  @Test
  void buildTreeTest() {
    var day = new Day07();
    String exampleInput = loadExample(2015, "day07.txt");

    day.buildTree(exampleInput);

    assertEquals(72, day.calculateValueForNode("d"));
    assertEquals(507, day.calculateValueForNode("e"));
    assertEquals(492, day.calculateValueForNode("f"));
    assertEquals(114, day.calculateValueForNode("g"));
    assertEquals(65412, day.calculateValueForNode("h"));
    assertEquals(65079, day.calculateValueForNode("i"));
    assertEquals(123, day.calculateValueForNode("x"));
    assertEquals(456, day.calculateValueForNode("y"));
  }
}

File: src/test/java/aoc2020/day07/Day07Test.kt
package aoc2020.day07

import aoc.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

class Day07Test {

    val input: String = loadExample(2020, "day07_1.txt")

    private lateinit var subject: Day07

    @BeforeEach
    fun setUp() {
        subject = Day07()
    }

    @Test
    fun part1Example() {
        assertThat(subject.part1(input))
            .isEqualTo(4)
    }

    @ParameterizedTest
    @CsvSource(
        "day07_1.txt, 32",
        "day07_2.txt, 126"
    )
    fun part2Example(filename: String, expectedResult: Int) {
        val input = loadExample(2020, filename)

        assertThat(subject.part2(input))
            .isEqualTo(expectedResult)
    }
}